app/src/main/java/com/polar/nextcloudservices/Notification/Processors/ActionsNotificationProcessor.java

```diff
@@ -33,37 +35,36 @@ public class ActionsNotificationProcessor implements AbstractNotificationProcess
     private static PendingIntent getCustomActionIntent(Context context, NotificationService service,
                                                        JSONObject action, int requestCode){
         Log.d(TAG, action.toString());
-        Intent intent = new Intent();
-        intent.setAction(Config.NotificationEventAction);
+        CustomTabsIntent browserIntent = new CustomTabsIntent.Builder()
+                .setUrlBarHidingEnabled(true)
+                .setShowTitle(false)
+                .setStartAnimations(context, android.R.anim.fade_in, android.R.anim.fade_out)
+                .setExitAnimations(context, android.R.anim.fade_in, android.R.anim.fade_out)
+                .setColorScheme(CustomTabsIntent.COLOR_SCHEME_SYSTEM)
+                .setShareState(CustomTabsIntent.SHARE_STATE_OFF)
+                .build();
+        browserIntent.intent.setAction(Config.NotificationEventAction);
+
         try {
-            intent.putExtra("notification_event", NOTIFICATION_EVENT_CUSTOM_ACTION);
+            browserIntent.intent.putExtra("notification_event", NOTIFICATION_EVENT_CUSTOM_ACTION);
             String link = action.getString("link");
             final String type = action.getString("type");
             link = Util.cleanUpURLIfNeeded(service.server, link);
             Log.d(TAG, link);
             Log.d(TAG, type);
-            intent.putExtra("action_link", link);
-            intent.putExtra("action_method", type);
+            browserIntent.intent.putExtra("action_link", link);
+            browserIntent.intent.putExtra("action_method", type);
         } catch (JSONException e) {
             //Log.d(TAG, action.toString());
             Log.e(TAG, "Can not get link or method from action provided by Nextcloud API");
             e.printStackTrace();
             return null;
         }
+
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
-            return PendingIntent.getBroadcast(
-                    context,
-                    requestCode,
-                    intent,
-                    PendingIntent.FLAG_MUTABLE | PendingIntent.FLAG_UPDATE_CURRENT
-            );
+            return PendingIntent.getActivity(context, requestCode, browserIntent.intent, PendingIntent.FLAG_MUTABLE | PendingIntent.FLAG_UPDATE_CURRENT);
         }else{
-            return PendingIntent.getBroadcast(
-                    context,
-                    requestCode,
-                    intent,
-                    PendingIntent.FLAG_UPDATE_CURRENT
-            );
+            return PendingIntent.getActivity(context, requestCode, browserIntent.intent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
Actually handling actions like this breaks 2-factor authentication via notification functionality( see issue #36). By default Nextcloud requires a CSRF token to be set to do any request, though obtaining CSRF token is not possible in app this why API is used. Quick actions in Nextcloud usually contain an API requests(though I know only two-factor auth via notification and Talk to use this feature). Below you may see example JSON for two-factor request actions:
```JSON
"actions": [
	{
		"label": "Схвалити",
		"link": "https://cloud.p01ar.net/ocs/v2.php/apps/twofactor_nextcloud_notification/api/v1/attempt/13",
		"type": "POST",
		"primary": true
	},
	{
		"label": "Скасувати",
		"link": "https://cloud.p01ar.net/ocs/v2.php/apps/twofactor_nextcloud_notification/api/v1/attempt/13",
		"type": "DELETE",
		"primary": false
	}
]
```

Actual clicking on this actions issue an API request instead of opening browser. Also I do not see where you handle method, but as far as I understand you are unable to make embedded browser to send specific type of request(`DELETE` or `POST`)

Reply:
Yes, you are right, thank you for your remarkes. I was actually not aware of the 2fa by notification app. So I will revert the changes for the `ActionsNotificationProcessor.java`. However, for my test instance the intent still not worked since there is a problem with the `cleanUpURLIfNeeded` in the `Util.java` file. I will open a new issue for this which would have to be fixed first...

Also, for calls the receive call or call back actions throw a `com.nextcloud.android.sso.exceptions.NextcloudUnsupportedMethodException`. So while the actions will work for 2fa notification app, they seem to be without function for nextcloud talk ...